pkg/sql/pgwire/pgerror/errors.go

```diff
@@ -132,6 +133,39 @@ func GetPGCause(err error) (*Error, bool) {
 	}
 }
 
+// NewAssertionErrorf creates an internal error.
+func NewAssertionErrorf(format string, args ...interface{}) error {
+	err := NewErrorWithDepthf(1, CodeInternalError, "programming error: "+format, args...)
+	err.InternalCommand = captureTrace()
+	err.Detail = err.InternalCommand
+	err.Hint = "You have encountered an error inside CockroachDB.\n\n" +
+		"Don't worry! Your data is likely safe.\n" +
+		"Please report this error with details at:\n\t https://github.com/cockroachdb/cockroach/issues/new/choose\nor\n\t support@cockroachlabs.com.\n\n" +
```

Comment:
Could mention searching for the unique error identifier on https://github.com/cockroachdb/cockroach/issues to see if it has known workarounds, etc.